suite/tests/linux/threadset.c

```diff
@@ -0,0 +1,353 @@
+        for (i = 0; i < NUM_SIMD_SSE_AVX_REGS; i++) {
+            for (j = 0; j < INTS_PER_YMM; j++)
+                buf[i * INTS_PER_YMM + j] = 0xdeadbeef + i * INTS_PER_ZMM + j;
+        }
+#    endif
+#    ifdef __AVX512F__
+#        define MOVE_TO_ZMM(buf, num)                           \
+            __asm__ __volatile__("vmovdqu64 %0, %%zmm" #num     \
+                                 :                              \
+                                 : "m"(buf[num * INTS_PER_ZMM]) \
+                                 : "zmm" #num);
+        MOVE_TO_ZMM(buf, 0)
+        MOVE_TO_ZMM(buf, 1)
+        MOVE_TO_ZMM(buf, 2)
+        MOVE_TO_ZMM(buf, 3)
+        MOVE_TO_ZMM(buf, 4)
+        MOVE_TO_ZMM(buf, 5)
+        MOVE_TO_ZMM(buf, 6)
+        MOVE_TO_ZMM(buf, 7)
+#        ifdef X64
+        MOVE_TO_ZMM(buf, 8)
+        MOVE_TO_ZMM(buf, 9)
+        MOVE_TO_ZMM(buf, 10)
+        MOVE_TO_ZMM(buf, 11)
+        MOVE_TO_ZMM(buf, 12)
+        MOVE_TO_ZMM(buf, 13)
+        MOVE_TO_ZMM(buf, 14)
+        MOVE_TO_ZMM(buf, 15)
+        MOVE_TO_ZMM(buf, 16)
+        MOVE_TO_ZMM(buf, 17)
+        MOVE_TO_ZMM(buf, 18)
+        MOVE_TO_ZMM(buf, 19)
+        MOVE_TO_ZMM(buf, 20)
+        MOVE_TO_ZMM(buf, 21)
+        MOVE_TO_ZMM(buf, 22)
+        MOVE_TO_ZMM(buf, 23)
+        MOVE_TO_ZMM(buf, 24)
+        MOVE_TO_ZMM(buf, 25)
+        MOVE_TO_ZMM(buf, 26)
+        MOVE_TO_ZMM(buf, 27)
+        MOVE_TO_ZMM(buf, 28)
+        MOVE_TO_ZMM(buf, 29)
+        MOVE_TO_ZMM(buf, 30)
+        MOVE_TO_ZMM(buf, 31)
+#        endif
+        /* Re-using INTS_PER_ZMM here to get same data patterns as above. */
+#        define MOVE_TO_OPMASK(buf, num)                        \
+            __asm__ __volatile__("kmovw %0, %%k" #num           \
+                                 :                              \
+                                 : "m"(buf[num * INTS_PER_ZMM]) \
+                                 : "k" #num);
+        MOVE_TO_OPMASK(buf, 0)
+        MOVE_TO_OPMASK(buf, 1)
+        MOVE_TO_OPMASK(buf, 2)
+        MOVE_TO_OPMASK(buf, 3)
+        MOVE_TO_OPMASK(buf, 4)
+        MOVE_TO_OPMASK(buf, 5)
+        MOVE_TO_OPMASK(buf, 6)
+        MOVE_TO_OPMASK(buf, 7)
+#    else
+#        define MOVE_TO_YMM(buf, num)                           \
+            __asm__ __volatile__("vmovdqu %0, %%ymm" #num       \
+                                 :                              \
+                                 : "m"(buf[num * INTS_PER_YMM]) \
+                                 : "ymm" #num);
+        MOVE_TO_YMM(buf, 0)
+        MOVE_TO_YMM(buf, 1)
+        MOVE_TO_YMM(buf, 2)
+        MOVE_TO_YMM(buf, 3)
+        MOVE_TO_YMM(buf, 4)
+        MOVE_TO_YMM(buf, 5)
+        MOVE_TO_YMM(buf, 6)
+        MOVE_TO_YMM(buf, 7)
+#        ifdef X64
+        MOVE_TO_YMM(buf, 8)
+        MOVE_TO_YMM(buf, 9)
+        MOVE_TO_YMM(buf, 10)
+        MOVE_TO_YMM(buf, 11)
+        MOVE_TO_YMM(buf, 12)
+        MOVE_TO_YMM(buf, 13)
+        MOVE_TO_YMM(buf, 14)
+        MOVE_TO_YMM(buf, 15)
```

Comment:
Hmm, there is a lot of duplicated lines here.  How about adding a `/* XXX: Try to share with sigcontext.c to avoid duplicating all the SIMD filling and checking code. */` up toward the top?